mailet/base/src/test/java/org/apache/mailet/base/test/FakeMailContext.java

```diff
@@ -44,11 +49,17 @@
         private final MailAddress sender;
         private final Collection<MailAddress> recipients;
         private final MimeMessage msg;
+        private final List<String> attributeNames;
 
-        public SentMail(MailAddress sender, Collection<MailAddress> recipients, MimeMessage msg) {
+        public SentMail(MailAddress sender, Collection<MailAddress> recipients, MimeMessage msg, List<String> attributeNames) {
             this.sender = sender;
-            this.recipients = recipients;
+            this.recipients = ImmutableList.copyOf(recipients);
             this.msg = msg;
+            this.attributeNames = Lists.newArrayList(attributeNames);
```

Comment:
SentMail represent a mail that was sent via FakeMailContext, so in tests.

Here the idea is to allow developper ensure the mail was sent with given attributes. So we need to store these attributes.